tfhe-zk-pok/src/curve_api/bls12_446.rs

```diff
@@ -431,51 +512,77 @@ mod g2 {
 }
 
 mod gt {
+    use crate::curve_446::{Fq12, Fq2};
+
     use super::*;
     use ark_ec::bls12::Bls12Config;
     use ark_ec::pairing::{MillerLoopOutput, Pairing};
-    use ark_ff::{CubicExtField, Fp12, Fp2, QuadExtField};
+    use ark_ff::{CubicExtField, QuadExtField};
 
     type Bls = crate::curve_446::Bls12_446;
     type Config = crate::curve_446::Config;
 
-    const ONE: Fp2<<Config as Bls12Config>::Fp2Config> = QuadExtField {
-        c0: MontFp!("1"),
-        c1: MontFp!("0"),
-    };
-    const ZERO: Fp2<<Config as Bls12Config>::Fp2Config> = QuadExtField {
+    const ZERO: Fq2 = QuadExtField {
         c0: MontFp!("0"),
         c1: MontFp!("0"),
     };
 
-    const U1: Fp12<<Config as Bls12Config>::Fp12Config> = QuadExtField {
-        c0: CubicExtField {
-            c0: ZERO,
-            c1: ZERO,
-            c2: ZERO,
-        },
-        c1: CubicExtField {
-            c0: ONE,
-            c1: ZERO,
-            c2: ZERO,
-        },
+    const TWO_INV: <Config as Bls12Config>::Fp = {
+        MontFp!("86412351771428577990035638289747981121746346761394949218917418178192828331138736448451251370148591845087981000773214233672031082665302")
     };
-    const U3: Fp12<<Config as Bls12Config>::Fp12Config> = QuadExtField {
-        c0: CubicExtField {
-            c0: ZERO,
-            c1: ZERO,
-            c2: ZERO,
-        },
-        c1: CubicExtField {
-            c0: ZERO,
-            c1: ONE,
-            c2: ZERO,
-        },
+    const TWO_INV_MINUS_1: <Config as Bls12Config>::Fp = {
+        MontFp!("86412351771428577990035638289747981121746346761394949218917418178192828331138736448451251370148591845087981000773214233672031082665301")
     };
 
-    const fn fp2_to_fp12(
-        x: Fp2<<Config as Bls12Config>::Fp2Config>,
-    ) -> Fp12<<Config as Bls12Config>::Fp12Config> {
+    const C: Fq2 = QuadExtField {
+        c0: TWO_INV,
+        c1: TWO_INV_MINUS_1,
+    };
+
+    fn fp2_mul_c(x: Fq2) -> Fq2 {
```

Comment:
some quick comments/docstrings on the functions could be welcome for other people who may contribute/want to use the functions for the ZK crate so that they can know what the primitives here do

Reply:
these are all internal details so they're not exposed in the api. the formulas here basically come from inlining the values of some constants given by Olivier to get rid of trivial multiplications and additions.